trucks.js

```diff
@@ -509,3 +511,65 @@ var foodTrucks = [
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
 // filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
 // filterByFoodType(foodType) - return trucks with associated 'foodType'
+
+
+// getTrucks() - return all trucks
+//function getTrucks(day) {
+//	var sortedTrucks = _.filter(foodTrucks, function (item) {
+//
+//		if (typeof item.schedule === 'undefined') {
+//			return false;
+//		} else {
+//			return (item.schedule.indexOf(day) !== -1);
+//		}
+//	});
+//
+//	return sortedTrucks;
+//}
+function getTrucks() {
+	return foodTrucks;
+}
+
+
+// getTruck(name) - return the truck object matching 'name'
+function getTruck(name) {
+
+	for(var i = 0, l = foodTrucks.length; i < l; i++) {
+
+		// Loop through food trucks until you find the matching name. When you find the name, stop.
+		if (foodTrucks[i].name.toLowerCase() === name.toLowerCase()) {
+
+			// return entire object for truck
+			return foodTrucks[i];
+		}
+	}
+
+	return false;
+}
+
+
+// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+function filterByDay(day) {
+	// Filter returns the subset of foodTrucks that match 'day'
+	var sortedTrucks = _.filter(foodTrucks, function (item) {
+
+		// Exclude truck if object doesn't include 'schedule' prop, otherwise check for 'day' match
+		if (typeof item.schedule === 'undefined') {
+			return false;
+		} else {
+			return (item.schedule.indexOf(day) !== -1);
+		}
+	});
+
+	return sortedTrucks;
+}
+
+
+console.log(getTruck('314 PIE'));
+console.log(getTrucks('Wednesday'));
+console.log(filterByDay('Tuesday'));
+
+
+module.exports = getTruck();
```

Comment:
These three statements aren't going to work like you expect them. These three lines each invoke a function and assign the result to module.exports. This means that when this module is done being required in by other code, it's going to be equal to the return value of calling filterByDay(), which is not what you want. What you want is to set module.exports equal to an object with each of these methods on it: `module.exports = { getTruck: getTruck, getTrucks, getTrucks, filterByDay: filterByDay };` Or, you could assign individual properties on module.exports to be equal to these same functions: `module.exports.getTruck = getTruck; module.exports.getTrucks = getTrucks; module.exports.filterByDay = filterByDay;` I recommend the first one - returning an object with methods set to the functions you've defined inside your module.